pkg/apiserver/handlers.go

```diff
@@ -367,16 +367,35 @@ func (r *requestAttributeGetter) GetAttribs(req *http.Request) authorizer.Attrib
 
 	attribs.ReadOnly = IsReadOnlyReq(*req)
 
-	apiRequestInfo, _ := r.apiRequestInfoResolver.GetAPIRequestInfo(req)
+	// Check whether meaningful api information can be resolved for the current path
+	reqHasApiPrefix := false
+	parts := splitPath(req.URL.Path)
+	i := 0
+	for range parts {
+		if r.apiRequestInfoResolver.APIPrefixes.Has(strings.Join(parts[i:], "/")) {
+			reqHasApiPrefix = true
+			break
+		}
+		i++
+	}
```

Comment:
I get nervous when I see code like this without accompanying tests. Should also test trailing slash handling (allowing `/api` vs `/api/`)